internal/sh/shell.go

```
@@ -752,6 +752,65 @@ func (shell *Shell) executeReturn(n *ast.ReturnNode) ([]sh.Obj, error) {
 	return returns, newErrStopWalking()
 }
 
+func (shell *Shell) getNashRootFromGOPATH(preverr error) (string, error) {
+	g, hasgopath := shell.Getenv("GOPATH")
+	if !hasgopath {
+		return "", errors.NewError("%s\nno GOPATH env var setted", preverr)
+	}
+	gopath := g.String()
+	return gopath + "/src/github.com/NeowayLabs/nash", nil
+}
+
+func isValidNashRoot(nashroot string) bool {
+	_, err := os.Stat(nashroot + "/stdlib")
+	return err == nil
+}
+
+func (shell *Shell) getNashRoot() (string, error) {
+	// TODO(katcipis): It is very annoying to load env vars, perhaps a shell.GetStringEnv ?
```

The problem is not existence checking, it is the type validation to see if the env var is a string or not, that is annoying =P. In the end I dont check and let it panic if the variable is not a string.